File: ios/ReactPOC/QuickLookThumbnailManager.m
//
//  QuickLookThumbnailManager.m
//  ReactPOC
//
//  Created by Geoffrey Xue on 7/27/20.
//

#import "ReactPOC-Bridging-Header.h"

@interface RCT_EXTERN_MODULE(QuickLookThumbnailManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(fileSource, NSNumber)
RCT_REMAP_VIEW_PROPERTY(url, urlString, NSString)
RCT_EXPORT_VIEW_PROPERTY(fileData, NSString)
RCT_EXPORT_VIEW_PROPERTY(fileType, NSString)
RCT_EXPORT_VIEW_PROPERTY(fileID, NSNumber)

RCT_EXPORT_VIEW_PROPERTY(onTap, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onLongPress, RCTDirectEventBlock)
@end

File: ios/ReactPOC/QuickLookThumbnailManager.swift
//
//  QuickLookThumbnailManager.swift
//  ReactPOC
//
//  Created by Geoffrey Xue on 7/27/20.
//

import Foundation


@objc(QuickLookThumbnailManager)
class QuickLookThumbnailManager : RCTViewManager {
  
  override static func requiresMainQueueSetup() -> Bool {
      return true
  }
  
  override func view() -> UIView! {
    return QuickLookThumbnail()
  }

}
